crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1918,18 +1922,22 @@ impl Default for WsHttpServers {
     }
 }
 
-/// Http Servers Enum
+// /// Http Servers Enum
+// enum WsHttpServerKind {
+//     /// Http server
+//     Plain(Server<Identity, RpcServerMetrics>),
+//     /// Http server with cors
+//     WithCors(Server<Stack<CorsLayer, Identity>, RpcServerMetrics>),
+//     /// Http server with auth
+//     WithAuth(Server<Stack<AuthLayer<JwtAuthValidator>, Identity>, RpcServerMetrics>),
+//     /// Http server with cors and auth
+//     WithCorsAuth(
+//         Server<Stack<AuthLayer<JwtAuthValidator>, Stack<CorsLayer, Identity>>, RpcServerMetrics>,
+//     ),
+// }
+
 enum WsHttpServerKind {
-    /// Http server
-    Plain(Server<Identity, RpcServerMetrics>),
-    /// Http server with cors
-    WithCors(Server<Stack<CorsLayer, Identity>, RpcServerMetrics>),
-    /// Http server with auth
-    WithAuth(Server<Stack<AuthLayer<JwtAuthValidator>, Identity>, RpcServerMetrics>),
-    /// Http server with cors and auth
-    WithCorsAuth(
-        Server<Stack<AuthLayer<JwtAuthValidator>, Stack<CorsLayer, Identity>>, RpcServerMetrics>,
-    ),
+    Server(Box<dyn Service<Request<Body>, Response = Response<Body>, Error = hyper::Error, Future = Pin<Box<dyn Future<Output = Result<Response<Body>, hyper::Error>>>>> + Send + Sync>),
```

Comment:
The type signature will be very big but we can flatten all of this into a bunch of `Either<Layer, Identiy>` which is what OptionLayer is:

https://docs.rs/tower/latest/tower/builder/struct.ServiceBuilder.html#method.option_layer